src/components/MoneyRequestHeader.tsx

```diff
@@ -1,89 +1,74 @@
-    report: iouReportPropTypes.isRequired,
-
-    /** The policy which the report is tied to */
-    policy: PropTypes.shape({
-        /** Name of the policy */
-        name: PropTypes.string,
-    }),
-
-    /** Personal details so we can get the ones for the report participants */
-    personalDetails: PropTypes.objectOf(participantPropTypes).isRequired,
-
-    /* Onyx Props */
+type MoneyRequestHeaderOnyxProps = {
     /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user email */
-        email: PropTypes.string,
-    }),
+    session: OnyxEntry<Session>;
 
     /** The expense report or iou report (only will have a value if this is a transaction thread) */
-    parentReport: iouReportPropTypes,
-
-    /** The report action the transaction is tied to from the parent report */
-    parentReportAction: PropTypes.shape(reportActionPropTypes),
+    parentReport: OnyxEntry<Report>;
 
     /** All the data for the transaction */
-    transaction: transactionPropTypes,
+    transaction: OnyxEntry<Transaction>;
+
+    /** All report actions */
+    // eslint-disable-next-line react/no-unused-prop-types
+    parentReportActions: OnyxEntry<ReportActions>;
 };
 
-const defaultProps = {
-    session: {
-        email: null,
-    },
-    parentReport: {},
-    parentReportAction: {},
-    transaction: {},
-    policy: {},
+type MoneyRequestHeaderProps = MoneyRequestHeaderOnyxProps & {
+    /** The report currently being looked at */
+    report: Report;
+
+    /** The policy which the report is tied to */
+    policy: Policy;
+
+    /** The report action the transaction is tied to from the parent report */
+    parentReportAction: ReportAction & OriginalMessageIOU;
+
+    /** Personal details so we can get the ones for the report participants */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 };
 
-function MoneyRequestHeader({session, parentReport, report, parentReportAction, transaction, policy, personalDetails}) {
+function MoneyRequestHeader({session, parentReport, report, parentReportAction, transaction, policy, personalDetails}: MoneyRequestHeaderProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
     const moneyRequestReport = parentReport;
-    const isSettled = ReportUtils.isSettled(moneyRequestReport.reportID);
+    const isSettled = ReportUtils.isSettled(moneyRequestReport?.reportID);
     const isApproved = ReportUtils.isReportApproved(moneyRequestReport);
     const {isSmallScreenWidth, windowWidth} = useWindowDimensions();
 
     // Only the requestor can take delete the request, admins can only edit it.
-    const isActionOwner = lodashGet(parentReportAction, 'actorAccountID') === lodashGet(session, 'accountID', null);
+    const isActionOwner = parentReportAction.actorAccountID === (session?.accountID ?? null);
```

Comment:
accountID's are of type int. Let's default to 0
```suggestion
    const isActionOwner = (parentReportAction?.actorAccountID ?? 0) === (session?.accountID ?? 0);
```

Reply:
@MonilBhavsar I believe the original idea was to default isActionOwner to false if one or both variables were defined.

If I apply your suggestion and both are undefined, then isActionOwner will be true which I don't think is the desired behaviour 😅

Reply:
Good point. Thanks for clarifying.
In that case, a better check would be to ensure both are integers, not undefined and then equal?